tests/contrib/hooks/test_gcp_api_base_hook.py

```diff
@@ -45,6 +45,57 @@
     default_creds_available = False
 
 
+class NoForbiddenAfterCount(object):
+    """Holds counter state for invoking a method several times in a row."""
+
+    def __init__(self, count, **kwargs):
+        self.counter = 0
+        self.count = count
+        self.kwargs = kwargs
+
+    def __call__(self):
+        """Raise an Forbidden until after count threshold has been crossed.
+        Then return True.
+        """
+        if self.counter < self.count:
+            self.counter += 1
+            raise Forbidden(**self.kwargs)
+        return True
+
+
+@hook.GoogleCloudBaseHook.quota_retry(wait=tenacity.wait_none())
+def _retryable_test_with_temporare_quota_retry(thing):
+    return thing()
+
+
+class retry_if_temporary_quotaTestCase(unittest.TestCase):
+    def test_do_nothing_on_non_error(self):
+        r = _retryable_test_with_temporare_quota_retry(lambda: 42)
+        self.assertTrue(r, 42)
+
+    def test_retry_on_exception(self):
+        message = "POST https://translation.googleapis.com/language/translate/v2: User Rate Limit Exceeded"
+        errors = [
+            {
+                'message': 'User Rate Limit Exceeded',
+                'domain': 'usageLimits',
+                'reason': 'userRateLimitExceeded',
+            }
+        ]
+        _retryable_test_with_temporare_quota_retry(NoForbiddenAfterCount(5, message=message, errors=errors))
```

Comment:
I am not sure if it is obvious what is tested here. Should adding  try / except improve it? what do you think?

Reply:
You're right. I will add try catch to emphasize intentions.